pkg/reconciler/managed/reconciler.go

```diff
@@ -1142,15 +1142,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (resu
 		}
 	}
 
-	if observation.ResourceUpToDate {
-		// We did not need to create, update, or delete our external resource.
-		// Per the below issue nothing will notify us if and when the external
-		// resource we manage changes, so we requeue a speculative reconcile
-		// after the specified poll interval in order to observe it and react
-		// accordingly.
-		// https://github.com/crossplane/crossplane/issues/289
+	// skip the update if the management policy is set to ignore updates
+	if !policy.ShouldUpdate() {
```

Comment:
This was originally lower in the reconciler, but we moved it here to catch cases where the management policy does not allow updates early.

Reply:
That is a good point! But I think I would rather rewrite the `if` on line 1157:
```
From this: if !observation.ResourceUpToDate && policy.ShouldUpdate() {
To this: if !observation.ResourceUpToDate {
```

And then I can immediately on the line below check if `!policy.ShouldUpdate()` and tell the user that the update will not be performed. This way it will only log this if an update is actually needed. 

Would this maybe create spam for the users, though? I am not very familiar with how the `ShouldUpdate` policy is set, but when it is set to false, would the user not be spammed with `Skipping update...` on every reconcile then?